### PR TITLE
Add a StdUnittest version

### PR DESCRIPTION
**Pull request by wilzbach**
There's still quite a bit of stuff in Phobos, that shouldn't be parsed & executed when Phobos is imported by a user who runs his tests.

For the motivation of excluding `unittest` in templates, please see https://wiki.dlang.org/DIP82

(I don't have any preferences on the name of this `version`.)

### CODE REVIEW

**Comment by CyberShadow** [comment]
Did we not have this or something very similar to this before? Am I thinking of another component?

**Comment by CyberShadow** [comment]
Oh, I was probably thinking of `StdDdoc`.

In which case maybe make the two consistent.

**Comment by JackStouffer** [comment]
bikesheading: any `version` you use should probably have two underscores prepended to it to avoid naming collisions with any existing code.

**Comment by wilzbach** [comment]
> Oh, I was probably thinking of StdDdoc.
> In which case maybe make the two consistent.

Renamed to `StdUnittest`, which @JackStouffer I think is unique enough and considering that this sadly will be used often in Phobos I would prefer to avoid anything with two ugly `__`.
We could add a changelog entry?

**Comment by wilzbach** [comment]
> I'd say let's just "reserve" all StdXyz identifiers by mentioning that in the Phobos docs, and then stick to that naming scheme.

Great idea!
-> https://github.com/dlang/dlang.org/pull/1999

File: std/experimental/allocator/building_blocks/bitmapped_block.d [hunk 1]
```diff
@@ -50,6 +50,7 @@ struct BitmappedBlock(size_t theBlockSize, uint theAlignment = platformAlignment
     import std.typecons : Ternary;
     import std.typecons : tuple, Tuple;
 
+    version(StdUnittest)
```
Comment (andralex) [inline std/experimental/allocator/building_blocks/bitmapped_block.d]: why do we need this here? I'd assume if no unittest is conducted, unittests are ignored anyway.

Comment (wilzbach) [reply std/experimental/allocator/building_blocks/bitmapped_block.d]: From DIP82:

> At present, unittest blocks inside of a templated type are compiled into every instantation, which only works with tests that are generic, which most tests aren't, and it almost never works with examples and thus doesn't work for ddoc-ed unittest blocks.

1) these unittest blocks might not even be run (I have a couple of these issues during the runnable examples story)
2) these unittest blocks will be executed as part of a user's testsuite

**Comment by wilzbach** [comment]
> I think we need here a version(unittest)

This already exists, but it's defined during a user's test run too.

**Comment by wilzbach** [comment]
FYI there have been two recent pull requests for which this identifier would have been useful:
- https://github.com/dlang/phobos/pull/5932 (already merged, I added an additional commit for it)
- https://github.com/dlang/phobos/pull/5951

So maybe we can simply move on with this? There are many other places in Phobos which should have used `StdUnittest` ...

**Pull request merged**